plugins/destination/s3/client/client.go

```diff
@@ -71,7 +72,7 @@ func New(ctx context.Context, logger zerolog.Logger, spec specs.Destination) (de
 	// we want to run this test because we want it to fail early if the bucket is not accessible
 	if _, err := c.uploader.Upload(ctx, &s3.PutObjectInput{
 		Bucket: aws.String(c.pluginSpec.Bucket),
-		Key:    aws.String("/tmp/.cq-test-file"),
+		Key:    aws.String(path.Join(spec.Path, "tmp", ".cq-test-file")),
```

Comment:
I believe this will need to perform some replacement of {{TABLE}} and {{UUID}}, otherwise it will write to a path that contains those literal strings